packages/@aws-cdk/assertions/lib/private/resources.ts

```diff
@@ -14,6 +14,41 @@ export function findResources(template: Template, type: string, props: any = {})
   return result.matches;
 }
 
+export function allResources(template: Template, type: string, props: any): string | void {
+  const section = template.Resources ?? {};
+  const result = matchSection(filterType(section, type), props);
+  if (result.match) {
+    if (result.mismatches) {
```

Comment:
How can a `result` both match and mismatch (naming might be decreasing the ease of reading)?

Why are there no tests for the new functionality?

Reply:
I took another look at this and decided to take `mismatches` out of the `MatchSuccess`. I agree it's confusing and after testing the functionality I realized I wasn't getting the information that I wanted in the error output. 

I reworked the functionality for easier readability and better messaging